nixos/doc/manual/development/option-types.section.md

```diff
@@ -63,6 +63,10 @@ merging is handled.
     ```
     :::
 
+`types.type`
+
+:   The type of a type itself.
```

Comment:
To understand what this is about, I had to read the source & test itself. While this is a rather niche thing, it'd still make sense to explain this a little bit more detailed here, right?

Reply:
Good point, I extended the description!